hl/test/test_image.c

```diff
@@ -665,10 +665,17 @@ test_generate(void)
     HL_TESTING2("make indexed image from sea data");
 
     for (i = 0; i < n_elements; i++) {
-        if (data[i] > 0)
+        if (data[i] > 0.0f)
             image_data[i] = 0;
-        else
-            image_data[i] = (unsigned char)((255 * (data[i] - xmin)) / xmin);
+        else {
+            assert(xmin < 0.0f); // never occurs
+
+            float temp = (255.0f * (data[i] - xmin)) / xmin;
+
+            assert(temp > -0.5f && temp < 255.5); // always occurs
+
+            image_data[i] = (unsigned char)(temp); // maybe call fabsf()?
+        }
```

Comment:
What UB was found in this function?

BTW, the error checking on the fscanf(3) calls looks wrong. `valex`, `xmin`, and `xmax` could all be undefined.

Reply:
As per the assert, `temp` is outside the range (-0.5, 255.5) thus converting it to uint8 is nonsensical.

Reply:
The expression assigned to `temp` on line 673 looks wrong to me. Probably should be `float temp = `255.0 * (data[i] - xmin) / (xmax - xmin);` if the intent is to map [xmin, xmax] to [0, 255].

Reply:
OK, test passes with that change, and UBSan is also happy.  Thanks!